alfasim_sdk/types.py

```diff
@@ -160,59 +363,284 @@ def check(self, attr: Attribute, value) -> None:
+
+    .. rubric:: Example using ``ALFAsimTypes`` on myplugin.py
+
+    .. code-block:: python
+
+        @data_model(icon="", caption="My Plugin")
+        class MyModel:
+            tracer_ref = MultipleReference(
+                ref_type=TracerType, caption="Tracer Type"
+            )
+
+
+    .. image:: _static/images/api/multiplereference_field_example_1.png
+
+    .. rubric:: Example using ``Custom Data`` on myplugin.py
+
+    .. code-block:: python
+
+        @data_model(caption="My Model")
+        class MyModel:
+            field_1 = String(value="Value 1", caption="String 1")
+
+
+        @container_model(caption="My Container", model=MyModel, icon="")
+        class MyContainer:
+            internal_ref = MultipleReference(
+                ref_type=MyModel,
+                container_type="MyContainer",
+                caption="Internal Reference",
+            )
+
+
+    .. image:: _static/images/api/multiplereference_field_example_2.png
+
+    .. rubric:: **Accessing MultipleReference Field from Plugin**:
+
+    In order to access this field from inside the plugin implementation, in C/C++,  you need to use :cpp:func:`get_plugin_input_data_multiplereference_selected_size`
+
+    .. rubric:: **Accessing MultipleReference Field from Context**:
+
+    When accessed from the :func:`~alfasim_sdk.context.Context`, the MultipleReference field will return a list with
+    the currently selected option objects instances.
+
+    With the instance, you can access all attributes from the object. Check the example below.
+
+    .. code-block:: bash
+
+        @data_model(caption="My Model")
+        class MyModel:
+            field_1 = String(value="Value 1", caption="String 1")
+
+
+        @container_model(caption="My Container", model=MyModel, icon="")
+        class MyContainer:
+            internal_ref = MultipleReference(
+                ref_type=MyModel,
+                container_type="MyContainer",
+                caption="Internal Reference",
+            )
+
+
+        # Example
+        >>> ctx.GetModel("MyContainer").internal_ref
+        [MyModel(field_1='Value 1', name='My Model 1'),
+        MyModel(field_1='Value 1', name='My Model 4')]
+
+        >>> type(ctx.GetModel("MyContainer").internal_ref)
+        <class 'list'>
+
+        >>> ctx.GetModel("MyContainer").internal_ref[0]
+        MyModel(field_1='Value 1', name='My Model 1')
+
     """
 
 
 @attr.s(kw_only=True, frozen=True)
 class Quantity(BaseField):
+    """
+    The Quantity field provides a way to the user provide a scalar value into the application.
+
+    The String fields have all options available from :func:`~alfasim_sdk.types.BaseField`, besides the listed the ones listed above:
+    :param values:  A number value.
+    :param unit:    Unit for the given scalar.
+
+    All scalar values are created using the `Barril library`_
+
+    Checkout the Barril documentation, to see all available units [ Needs to implement on Barril this section]
```

Comment:
What does "[ Needs to implement on Barril this section]" means?

Reply:
We need something on the barril library/documentation to show all possible units.

This behavior is not documented anywhere.

Reply:
I see, but update this text to something better.

Reply:
Updated with a link for the barril documentation section that has all available units ;)
![image](https://user-images.githubusercontent.com/5083518/67510137-c7c42e80-f66a-11e9-82bc-902cb7941a0c.png)